### PR TITLE
Create selection_sort.rb

### PR DESCRIPTION
**Pull request by cy-by-side**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.

### CODE REVIEW

File: codes/ruby/selection_sort.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+=begin
+File: selection_sort.rb
+Created Time: 2024-04-02
+Author: Cy (9738314@gmail.com)
+=end
+
+### 选择排序 ###
+def selection_sort(nums)
+  n = nums.length()
```
Comment (khoaxuantu) [inline codes/ruby/selection_sort.rb:9]:
```suggestion
  n = nums.length
```

File: codes/ruby/selection_sort.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  # 外排序：未排序区间为[i, n-1]
+  for i in 0...n - 1
```
Comment (khoaxuantu) [inline codes/ruby/selection_sort.rb:11]: We should write the range in for loop as the following for better readability: 

```suggestion
  for i in 0...(n - 1)
```

File: codes/ruby/selection_sort.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    k = i
+    # 内循环：找到未排序区间内的最小元素
+    for j in i + 1...n
```
Comment (khoaxuantu) [inline codes/ruby/selection_sort.rb:14]:
```suggestion
    for j in (i + 1)...n
```

**Review by khoaxuantu — COMMENTED** [review]
Hi @cy-by-side, I have some comments

File: codes/ruby/selection_sort.rb (hunk 3 above)
Comment (cy-by-side) [reply codes/ruby/selection_sort.rb:14]: thx, i will fix these

File: codes/ruby/chapter_sorting/selection_sort.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+=begin
+File: selection_sort.rb
+Created Time: 2024-04-02
+Author: Cy (3739004@gmail.com) Xuan Khoa Tu Nguyen (ngxktuzkai2000@gmail.com)
```
Comment (krahets) [inline codes/ruby/chapter_sorting/selection_sort.rb:4]: Add a comma between the two ids.

File: codes/ruby/selection_sort.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
+=begin
```
Comment (krahets) [inline codes/ruby/selection_sort.rb:1]: Redundant file

**Pull request closed without merging**